src/server/scripts/Northrend/Naxxramas/boss_grobbulus.cpp

```diff
@@ -15,35 +15,48 @@
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "ScriptMgr.h"
+/*
+ * He should call all stitched giants after aggro
+ */
+
+#include "naxxramas.h"
+#include "ObjectAccessor.h"
+#include "PassiveAI.h"
 #include "ScriptedCreature.h"
-#include "SpellScript.h"
+#include "ScriptMgr.h"
 #include "SpellAuraEffects.h"
-#include "naxxramas.h"
+#include "SpellScript.h"
 
-enum Spells
+enum GrobbulusTexts
 {
-    SPELL_BOMBARD_SLIME         = 28280,
-    SPELL_SLIME_SPRAY           = 28157,
+    EMOTE_SLIME                 = 0,
+    EMOTE_MUTAGEN               = 1
+};
+
+enum GrobbulusSpells
+{
+    // Grobbulus
+    SPELL_BOMBARD_SLIME         = 28280,   // should be used out of combat (waypoint script?)
+    SPELL_SUMMON_FALLOUT_SLIME  = 28218,
     SPELL_MUTATING_INJECTION    = 28169,
     SPELL_MUTATING_EXPLOSION    = 28206,
     SPELL_POISON_CLOUD          = 28240,
+    SPELL_BERSERK               = 26662,
+    // Grobbulus Cloud
     SPELL_POISON_CLOUD_PASSIVE  = 28158,
     SPELL_PACIFY_SELF           = 19951,
-    SPELL_BERSERK               = 26662
+    // Fallout Slime
+    SPELL_DISEASE_CLOUD         = 54367
 };
 
-enum Events
-{
-    EVENT_BERSERK               = 1,
-    EVENT_CLOUD                 = 2,
-    EVENT_INJECT                = 3,
-    EVENT_SPRAY                 = 4
-};
+#define SPELL_SLIME_SPRAY RAID_MODE<uint32>(28157, 54364)
```

Comment:
`sSpellMgr->GetSpellIdForDifficulty(SPELL_UNBOUND_PLAGUE, GetCaster())``

this is what I meant

Reply:
Nice thing, thanks